<a href="https://colab.research.google.com/github/itsGab/ds_febre_amarela/blob/main/notebooks/analise_exploratoria.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<h1 align='center'> Previsão de Admissão na UTI por COVID-19 </h1>
<h2 align='center'> Notebook De Análise Exploratória </h2>

# IMPORTAÇÕES (data & libs)

In [1]:
# imports
import pandas as pd

# importação dos dados
data_raw = 'https://github.com/itsGab/previsao_uti_em_covid/blob/main/dados/Kaggle_Sirio_Libanes_ICU_Prediction.xlsx?raw=true'

# ABRINDO O CONJUNTO DE DADOS


Este **conjunto de dados** contém dados anônimos do Hospital Sírio-Libanês, São Paulo e Brasília. Todos os dados foram tornados anônimos de acordo com as melhores práticas e recomendações internacionais. Os dados foram limpos e escalados por coluna de acordo com o Min Max Scaler para caber entre -1 e 1.


In [2]:
# lendo os dados com pandas
df = pd.read_excel(data_raw)

# removendo espaco no nome das colunas
df.columns = df.columns.str.replace(' ', '_')

# mostrando o formato e tamanho do dataframe
print(f'formato: {df.shape}')
print(f'tamanho: {df.size}')

# mostrando o cabeçalho do dataframe
df.head()

formato: (1925, 231)
tamanho: 444675


PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65  ...    WINDOW  ICU
0                         0            1  ...       0-2    0
1                         0            1  ...       2-4    0
2                         0            1  ...       4-6    0
3                         0            1  ...      6-12    0
4                         0            1  ...  ABOVE_12    1

[5 rows x 231 columns]

Nessa exibição inicial é possível ver o formato do dataframe, com o qual estamos trabalhando 

| Formato | Descrição |
|--|--|
| Linhas | 1925 |
| Colunas | 231 |

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1925 entries, 0 to 1924
Columns: 231 entries, PATIENT_VISIT_IDENTIFIER to ICU
dtypes: float64(225), int64(4), object(2)
memory usage: 3.4+ MB


<h3 align='center'> Informações: </h3>

| Info | Descrição |
|--|--|
| Nº de colunas | 231 |
| Nº de linhas | 1925 | 
| Data types| ["float64(225)", "int64(4)", "object(2)"] |
| Nº de pacientes | 385 | 

# VALORES FALTANTES

### Porcentagm total de valores faltantes

In [4]:
df.isnull().sum().sum()/df.size

0.5034305953786473

Os valores faltantes (ou vazios, NaN), representam 50.34 % da nossa base de dados.

### Quantidade de valores faltantes por colunas

In [5]:
df.isnull().sum().sort_values(ascending=False)

P02_VENOUS_MIN              1104
HEMOGLOBIN_MEAN             1104
LACTATE_MIN                 1104
LACTATE_MEAN                1104
LACTATE_MEDIAN              1104
                            ... 
AGE_ABOVE65                    0
ICU                            0
GENDER                         0
WINDOW                         0
PATIENT_VISIT_IDENTIFIER       0
Length: 231, dtype: int64

# SEPARAÇÃO DE COLUNAS POR TIPO

As colunas podem ser categorizadas em 5 principais tipos, sendo elas: 

* Colunas informativas: são essa o identificador do paciente, a janela de tempo e se o paciente foi ou não admitido na UTI. 

* Colunas de demograficas: representam informações como sexo ou idade do paciente.

* Colunas de comorbidades: representam algum grupo de doença ou comorbidades que o paciente venha ou nao a ter.

* Colunas de sinais vitais: representam resultados de sinais vitais do paciente

* Colunas de exames laboratoriais: representam resultados de exames laboratorias feitos no paciente.



Fazendo a separação das colunas em lista por tipo e salvando em CSV para consulta

In [6]:
# Baseado em um notebook feito pelo criador do desafio
# https://www.kaggle.com/felipeveiga/starter-covid-19-sirio-libanes-icu-admission

# informativas
cols_informativas = ["PATIENT_VISIT_IDENTIFIER", "WINDOW", "ICU"]

# demograficas
cols_demograficas = [i for i in df.columns if "AGE_" in i]
cols_demograficas.append("GENDER")

# comorbidades
cols_comorbidades = [i for i in df.columns if "DISEASE" in i]
cols_comorbidades.extend(["HTN", "IMMUNOCOMPROMISED", "OTHER"])

# sinais vitais
cols_sinais_vitais = df.iloc[:,193:-2].columns.tolist()

# exames laboratoriais
cols_exames_laboratoriais = df.iloc[:,13:193].columns.tolist()

# criando uma Series para consulta
s_colunas = pd.Series(data={'informativas': cols_informativas,
                            'demograficas': cols_demograficas,
                            'comorbidades': cols_comorbidades,
                            'sinais_vitais': cols_sinais_vitais,
                            'exames_laboratoriais': cols_exames_laboratoriais}, 
                      name = 'Colunas_por_tipo')

# salvando em arquivo CSV
s_colunas.to_csv('Colunas_por_tipo')

# COLUNAS INFORMATIVAS

## COLUNAS

In [7]:
df[cols_informativas][:7]

PATIENT_VISIT_IDENTIFIER    WINDOW  ICU
0                         0       0-2    0
1                         0       2-4    0
2                         0       4-6    0
3                         0      6-12    0
4                         0  ABOVE_12    1
5                         1       0-2    1
6                         1       2-4    1

As colunas informativas são compostas:

| Colunas | Descrição |
| -- | -- |
| PATIENT_VISIT_IDENTIFIER | Identificador do paciente |
| WINDOW| Janela de tempo |
| ICU | Variável de admissão na UTI |

### **IDENTIFICADOR DE PACIENTE**

In [8]:
df['PATIENT_VISIT_IDENTIFIER'].nunique()

385

A Coluna '**PATIENT_VISIT_IDENTIFIER**' é composta por **385** pacientes de identificador único, indo de 0 à 384. Cada identificador **aparece** 5 vezes, dentro do conjunto de dados, devido as **5 janelas de admissão**.

### **WINDOW**

In [9]:
df['WINDOW'].unique()

array(['0-2', '2-4', '4-6', '6-12', 'ABOVE_12'], dtype=object)

A Coluna **'WINDOW'** foi divida em 5 janelas de admissão na UTI.

| Window | Descrição |
|--|--|
| 0-2 | De 0 á 2 horas da admissão |
| 2-4 | De 2 a 4 horas da admissão |
| 4-6 | De 4 a 6 horas da admissão |
| 6-12 | De 6 a 12 horas da admissão |
| Above_12 | Acima de 12 horas da admissão |

### **UTI**

In [10]:
df['ICU'].unique()

array([0, 1])

A variável **UTI**, representada por **'ICU'**, é a nossa variável alvo, ou seja, a variável da qual queremos prever no modelo de *Machine Learning*. 

Seus valores são booleanos (Verdadeiro: 1, Falso: 0), representando a admissão ou não admissão na UTI.

| ICU | Descrição |
|--|--|
| 0 | Não teve admissão na UTI |
| 1 | Teve admissão na UTI |

# COLUNAS DEMOGRÁFICAS


## COLUNAS

In [11]:
df[cols_demograficas].head()

AGE_ABOVE65 AGE_PERCENTIL  GENDER
0            1          60th       0
1            1          60th       0
2            1          60th       0
3            1          60th       0
4            1          60th       0

As colunas demográficas são compostas:

| Colunas | Descrição |
| -- | -- |
| AGE_ABOVE65 | Acima de 65 anos |
| AGE_PERCENTIL | Percentil de idade |
| GENDER | Sexo (masculino ou feminino) |

### **IDADE ACIMA DE 65 ANOS**


In [12]:
df['AGE_ABOVE65'].unique()

array([1, 0])

|AGE_ABOVE65|Descrição|
|--|--|
|0|Idade menor ou igual a 65 anos|
|1|Idade acima de 65 anos|

### **PERCENTIS DE IDADE**



In [13]:
sorted(df['AGE_PERCENTIL'].unique().tolist())

['10th',
 '20th',
 '30th',
 '40th',
 '50th',
 '60th',
 '70th',
 '80th',
 '90th',
 'Above 90th']

### **GÊNERO**


In [14]:
df['GENDER'].unique()

array([0, 1])

| GENDER | Descrição |
|--|--|
| 1 | Feminino |
| 0 | Masculino |

# COLUNAS DE COMORBIDADES

## COLUNAS

In [15]:
df[cols_comorbidades]

DISEASE_GROUPING_1  DISEASE_GROUPING_2  ...  IMMUNOCOMPROMISED  OTHER
0                    0.0                 0.0  ...                0.0    1.0
1                    0.0                 0.0  ...                0.0    1.0
2                    0.0                 0.0  ...                0.0    1.0
3                    0.0                 0.0  ...                0.0    1.0
4                    0.0                 0.0  ...                0.0    1.0
...                  ...                 ...  ...                ...    ...
1920                 0.0                 0.0  ...                0.0    1.0
1921                 0.0                 0.0  ...                0.0    1.0
1922                 0.0                 0.0  ...                0.0    1.0
1923                 0.0                 0.0  ...                0.0    1.0
1924                 0.0                 0.0  ...                0.0    1.0

[1925 rows x 9 columns]

As colunas identificadas como comorbidades, tem valores _booleanos_, identificando se o paciente, em questão, tem ou não a comorbidade.

| Valor | Descrição |
|--|--|
| 1 | Possui (Verdadeiro)|
| 0 | Não possui (Falso) |

### DESCRIÇÃO

| COMORBIDADES | Descrição|
|--|--|
| DISEASE_GROUPING | Doenças Agrupadas |
| HTN | Hipertensão |
| IMMUNOCOMPROMISED | Imunocomprometido |
| OTHER | Outras |

### DOENÇAS AGRUPADAS

In [16]:
d_agrupadas = [col for col in df[cols_comorbidades].columns if 'DISEASE_GROUPING' in col]
d_agrupadas

['DISEASE_GROUPING_1',
 'DISEASE_GROUPING_2',
 'DISEASE_GROUPING_3',
 'DISEASE_GROUPING_4',
 'DISEASE_GROUPING_5',
 'DISEASE_GROUPING_6']

Para melhorar a privacidade de paciente é feita uma agrupamento das doenças. Esses agrupamento foram divididos em 6 grupos sem identificação das doenças contidas neles.

# COLUNAS SINAIS VITAIS E EXAMES LAB.

## INFORMAÇÕES

|Colunas| Quantidade|
|--|--|
|Sinais Vitiais| 6 |
|Exames Laboratoriais| 36 |

As colunas de exames são expandidas em **média, mediana, máxima, mínima, diferença e diferença relativa**.

Os números forma escalados por coluna de acordo com o Min Max Scaler para caber entre **-1 e 1**.

## SINAIS VITAIS

As colunas de **sinais vitais** são divida em 6 diferentes, no total são 36 colunas com suas variações.

In [17]:
df[cols_sinais_vitais].head()

BLOODPRESSURE_DIASTOLIC_MEAN  ...  OXYGEN_SATURATION_DIFF_REL
0                      0.086420  ...                   -1.000000
1                      0.333333  ...                   -1.000000
2                           NaN  ...                         NaN
3                           NaN  ...                   -1.000000
4                     -0.243021  ...                   -0.814433

[5 rows x 36 columns]

In [18]:
df[cols_sinais_vitais].describe()

BLOODPRESSURE_DIASTOLIC_MEAN  ...  OXYGEN_SATURATION_DIFF_REL
count                   1240.000000  ...                 1239.000000
mean                      -0.093631  ...                   -0.886982
std                        0.252064  ...                    0.296772
min                       -1.000000  ...                   -1.000000
25%                       -0.262708  ...                   -1.000000
50%                       -0.100172  ...                   -0.980333
75%                        0.086420  ...                   -0.880155
max                        1.000000  ...                    1.000000

[8 rows x 36 columns]

## EXAMES LABORATORIAIS

As colunas de **exames laboratoriais** são divida em 36 diferentes, no total são 180 colunas com suas variações.

In [19]:
df[cols_exames_laboratoriais].head()

ALBUMIN_MEDIAN  ALBUMIN_MEAN  ALBUMIN_MIN  ...  DIMER_MIN  DIMER_MAX  DIMER_DIFF
0             NaN           NaN          NaN  ...        NaN        NaN         NaN
1             NaN           NaN          NaN  ...        NaN        NaN         NaN
2        0.605263      0.605263     0.605263  ...  -0.994912  -0.994912        -1.0
3             NaN           NaN          NaN  ...        NaN        NaN         NaN
4        0.000000      0.000000     0.000000  ...  -0.996762  -0.996762        -1.0

[5 rows x 180 columns]

In [20]:
df[cols_exames_laboratoriais].describe()

ALBUMIN_MEDIAN  ALBUMIN_MEAN  ...   DIMER_MAX  DIMER_DIFF
count      821.000000    821.000000  ...  821.000000       821.0
mean         0.528527      0.528527  ...   -0.954177        -1.0
std          0.224100      0.224100  ...    0.123582         0.0
min         -1.000000     -1.000000  ...   -1.000000        -1.0
25%          0.605263      0.605263  ...   -0.978877        -1.0
50%          0.605263      0.605263  ...   -0.978029        -1.0
75%          0.605263      0.605263  ...   -0.968315        -1.0
max          1.000000      1.000000  ...    1.000000        -1.0

[8 rows x 180 columns]